src/model/SequenceView.tsx

```diff
@@ -0,0 +1,222 @@
+import fetchData from '../utils/fetchData';
+import { formatLargeNumber } from '../utils/utils';
+import idx from 'idx';
+
+enum Flag {
+  PRECURSOR = 'Precursor',
+  FRAGMENT = 'Fragment',
+  FRAGMENTS = 'Fragments',
+  FRAGMENT_PRECURSOR = 'Fragment,Precursor',
+  FRAGMENTS_PRECURSOR = 'Fragments,Precursor',
+}
+
+type Isoform = {
+  name: { value: string };
+  isoformSequenceStatus: string;
+  isoformIds: string[];
+  synonyms: Array<{ value: string }>;
+  note: { texts: Array<{ value: string }> };
+  sequenceIds: string[];
+};
+
+type AlternativeProduct = {
+  commentType: FreeTextType.ALTERNATIVE_PRODUCTS;
+  isoforms: Isoform[];
+  note: { texts: Array<{ value: string }> };
+  events: string[];
+};
+
+type SequenceViewProps = {
+  data: {
+    entryAudit?: {
+      lastSequenceUpdateDate?: string;
+      sequenceVersion: string;
+    };
+    proteinDescription?: {
+      flag?: Flag;
+    };
+    sequence?: {
+      value: string;
+      length: number;
+      molWeight: number;
+      crc64: string;
+    };
+    comments?: AlternativeProduct[];
+  };
+};
+
+export const SequenceInfo: React.FC<{
+  isoformId: string;
+  isoformSequence?: SequenceViewProps['data']['sequence'];
+  lastUpdateDate?: string | null;
+}> = ({ isoformId, isoformSequence, lastUpdateDate }) => {
+  const [data, setData] = useState(null);
+  const [isoformToFetch, setIsoformToFetch] = useState('');
+
+  useEffect(() => {
+    if (!isoformToFetch) {
+      return;
+    }
+    const fetchIsoformData = async () => {
+      const result = await fetchData(`${apiUrls.entry(isoformToFetch)}`);
+      setData(result.data.sequence);
+    };
+
+    fetchIsoformData();
+  }, [isoformToFetch]);
+
+  const dataToDisplay = data ? data : isoformSequence;
+
+  if (!dataToDisplay) {
+    return (
+      <button
+        type="button"
+        className="button"
+        onClick={() => setIsoformToFetch(isoformId)}
+      >
+        Load sequence
+      </button>
+    );
+  }
+  const infoData = [
+    {
+      title: 'Length',
+      content: dataToDisplay.length,
+    },
+    {
+      title: 'Mass (Da)',
+      content: formatLargeNumber(dataToDisplay.molWeight),
+    },
+    {
+      title: 'Last updated',
+      content: lastUpdateDate,
+    },
+    {
+      title: 'Checksum',
+      content: dataToDisplay.crc64,
+    },
+  ];
+  return (
+    <Fragment>
+      {dataToDisplay && <InfoList infoData={infoData} />}
+      <Sequence sequence={dataToDisplay.value} />
+    </Fragment>
+  );
+};
+
+export const IsoformInfo: React.FC<{ isoformData: Isoform }> = ({
+  isoformData,
+}) => {
+  const infoListData = [
+    {
+      title: 'Name',
+      content: isoformData.name.value,
+    },
+    {
+      title: 'Synonyms',
+      content:
+        isoformData.synonyms &&
+        isoformData.synonyms.map(syn => syn.value).join(', '),
+    },
+    {
+      title: 'Note',
+      content:
+        isoformData.note &&
+        isoformData.note.texts.map(note => note.value).join(', '),
```

Comment:
Could use `idx` since it's already imported.

Reply:
Same as above